Tests/ARTFallbackTest.m

```diff
@@ -43,4 +43,35 @@ - (void)testAllHostsIncludedOnce {
     }    
 }
 
+- (void)testCustomFallbackHosts {
+    NSArray *customHosts = @[@"test.ably.com", @"test2.ably.com", @"test3.ably.com"];
+    ARTFallback *f = [[ARTFallback alloc] initWithFallbackHosts:customHosts];
+    
+    NSSet *customSet = [NSSet setWithArray:customHosts];
+    NSMutableArray *hostsRandomised = [NSMutableArray array];
+    for(int i=0;i < [customHosts count]; i++) {
+        [hostsRandomised addObject:[f popFallbackHost]];
+    }
+    
+    //popping after all hosts are exhausted returns nil
+    XCTAssertTrue([f popFallbackHost] == nil);
+    
+    // all fallback hosts are used in artfallback
+    XCTAssertEqual([hostsRandomised count], [customHosts count]);
+    bool inOrder = true;
+    for(int i=0;i < [customHosts count]; i++) {
+        if(![[customHosts objectAtIndex:i] isEqualToString:[hostsRandomised objectAtIndex:i]]) {
+            inOrder = false;
+            break;
+        }
+    }
+    //check artfallback randomises the order.
+    XCTAssertFalse(inOrder);
```

Comment:
We have already support for doing this clean by overriding the random function. See https://github.com/ably/ably-ios/blob/e049dab0e87eaa1643e21d70ae3a771529e0d9e7/Spec/RealtimeClientConnection.swift#L2362-L2372 which lets us expect a given order: https://github.com/ably/ably-ios/blob/e049dab0e87eaa1643e21d70ae3a771529e0d9e7/Spec/RealtimeClientConnection.swift#L2585 If we get the expected order, we prove that fallback logic is using the `ARTFallback_getRandomHostIndex` function correctly, and we can just trust that the default value for `ARTFallback_getRandomHostIndex` works as expected